File: Back/NodeJS/src/domain/answerBets/AnswerBet.ts
import { BetId } from "../bets/BetId";
import { Entity } from "../Entity";
import { MemberId } from "../members/MemberId";
import { BetAnswered } from "./events/BetAnswered";

export class AnswerBet extends Entity {
    constructor(public readonly BetId: BetId, 
                public readonly Answer: boolean,
                public readonly GamblerId: MemberId) {
        super()
        this.AddDomainEvent(new BetAnswered(this.BetId, this.GamblerId, this.Answer))
    }
}

File: Back/.Net/BetFriends.Domain/AnswerBets/AnswerBet.cs

using BetFriends.Domain.Bets;
using BetFriends.Domain.Events;
using BetFriends.Domain.Members;

namespace BetFriends.Domain.AnswerBets;

public class AnswerBet : Entity
{
    private readonly BetId betId;
    private readonly bool answer;
    private readonly MemberId memberId;

    public AnswerBet(BetId betId, bool answer, MemberId memberId)
    {
        this.betId = betId;
        this.answer = answer;
        this.memberId = memberId;
        AddEvent(new BetAnswered(betId, answer, memberId));
    }

    public AnswerBetState State { get => new AnswerBetState(betId, memberId, answer); }
}
